apps/dashboard/src/app/(internal)/event/components/qr-code-scanner.tsx

```diff
@@ -14,6 +14,7 @@ interface QrCodeScannerProps {
 export const QrCodeScanner: FC<QrCodeScannerProps> = ({ attendance }) => {
   const [scannerOpen, { toggle: toggleScanner }] = useDisclosure(false)
   const [videoReady, setVideoReady] = useState(false)
+  const paused = useRef(false)
```

Comment:
Is there a particular reason this is a ref and not a state?

Reply:
Yes. Calling useState from the modal gives an error, since a component is trying to update another component